### ISSUE TITLE
Starting with a null model, can't choose "Today"

### ISSUE THREAD

**Issue by kijanawoodard** [fragaria/angular-daterangepicker#86]
The problem we're seeing is that if the user clears the date manually, the control still thinks the previous date is still selected and won't let you select it again.

In the simple case, if the model is set to null, you can't pick "today" because it's already selected in the control.

**Comment by belerweb** [comment]
I got the same issue.

**Comment by atardadi** [comment]
+1

**Comment by asabirov** [comment]
+1

**Comment by jgolebiewski** [comment]
+1

**Comment by Till86** [comment]
+1 anyone made progress on the issue?

**Comment by Till86** [comment]
Guess it was related to moment.js setting date to now, when invalid value was entered. should be fixed now (moment.js 2.11.0), so maybe there is a way to fix this issue, too.

**Comment by drelei** [comment]
I used "moment().toISOString()" for the value as a workaround.

**Comment by cgatesman** [comment]
FWIW, as a workaround, I added the following code into angular-daterangepicker in `_init()` after creating the daterangepicker:

```javascript
          el.on('apply.daterangepicker', function() {
            if(opts.singleDatePicker) {
              if($scope.model == null) {
                return $scope.$apply(function() {
                  return $scope.model = moment().startOf('day');
                });
              }
            } else if(angular.isObject($scope.model) && $scope.model.startDate == null && $scope.model.endDate == null) {
              return $scope.$apply(function() {
                return $scope.model = {
                  startDate: moment().startOf('day'),
                  endDate: moment().endOf('day')
                };
              });
            }
          });
```

**Comment by cgatesman** [comment]
Curious...why was this issue closed?  Was this fixed in 3.0?  I couldn't find a merged PR referencing this issue.
